core/appeal/service.go

```diff
@@ -1121,6 +1134,110 @@ func getPolicy(a *domain.Appeal, p *domain.Provider, policiesMap map[string]map[
 	return policy, nil
 }
 
+func (s *Service) populateAppealMetadata(ctx context.Context, a *domain.Appeal, p *domain.Policy) error {
+	if p.AppealMetadataSources == nil {
+		return nil
+	}
+
+	appealMetadata := map[string]interface{}{}
+	for key, metadata := range p.AppealMetadataSources {
+		if metadata.Type == "http" {
+			if metadata.Config.URL == "" {
+				return fmt.Errorf("URL cannot be empty for http type")
+			}
+
+			metadataCl, err := s.getMetadataClient(metadata.Config)
+			if err != nil {
+				return fmt.Errorf("key: %s, %w", key, err)
+			}
+
+			res, err := metadataCl.GetResourceMetadata()
+			if err != nil && !metadata.Config.AllowFailed {
+				return fmt.Errorf("error fetching resource: %w", err)
+			}
+
+			responseMap := res.(map[string]interface{})
+			params := map[string]interface{}{
+				"response": responseMap,
```

Comment:
let's expose the original response values instead of only the body so user can access `response.status`, `response.headers`, `response.body`, etc.